src/store/actions/thunk.ts

```diff
@@ -0,0 +1,48 @@
+import { Dispatch } from '@reduxjs/toolkit';
+import axios, { AxiosError } from 'axios';
+import { Videos } from '../../components/VideoList/types';
+import { BASE_FETCH_URL, API_KEY, MAX_RESULTS } from '../../constants/api';
+import { errorActions } from '../slices/error';
+import { loadingActions } from '../slices/loading';
+import { videosActions } from '../slices/videos';
+
+export function getVideos(
```

Comment:
what this is for?

Reply:
type overload? I've read somewhere it's much more specific than using an optional parameter.